test/lib/jdk/test/lib/apps/LingeredApp.java

```diff
@@ -495,6 +495,45 @@ public static boolean isLastModifiedWorking() {
         return sane;
     }
 
+    /**
+     * Support for creating a thread whose stack trace is always readable. There are
+     * occassional isues trying to get the stack trace for LingeredApp.main() since
+     * it sometimes wakes up from sleep and may breifly have an unreadable thread
+     * stack trace. The code below is used to create "SteadyStateThread" whose
+     * stack trace is always readable.
+     */
+
+    private static volatile boolean steadyStateReached = false;
+
+    private static void steadyState(Object steadyStateObj) {
+        steadyStateReached = true;
+        synchronized(steadyStateObj) {
+        }
+    }
+
+    private static void startSteadStateThread(Object steadyStateObj) {
+        Thread steadyStateThread = new Thread() {
+            public void run() {
+                steadyState(steadyStateObj);
+            }
+        };
+        steadyStateThread.setName("SteadyStateThread");
+        steadyStateThread.start();
+
+        try {
+            // Sleep until the thread has started running.
+            while (!steadyStateReached) {
+                Thread.sleep(100);
+            }
+            // Do another short sleep so we can get into the synchronized block,
+            // although this probably is not necessary to guarantee that the
+            // stack trace is readable.
+            Thread.sleep(100);
```

Comment:
Can we wait to change the state of `steadyStateThread` to `BLOCKED`? It is more robustness than `Thread.sleep(100)`.

Reply:
How do you do that?

Reply:
For example, can we write following code?

```
while (steadyStateThread.getState() != Thread.State.BLOCKED) {
    Thread.onSpinWait();
}
```

Reply:
Ok, that's easy enough. What about the loop before it. Do you prefer 100ms sleeps or onSpinWait() for it also.

Reply:
I prefer `onSpinWait()` if anything because it expects short suspend, and we can expect CPU friendly code.

Reply:
As an option, we can remove `steadyStateReached` and rewrite the code as following:

```
        steadyStateThread.setName("SteadyStateThread");
        steadyStateThread.start();

        while (steadyStateThread.getState() != Thread.State.BLOCKED) {
            Thread.onSpinWait();
        }
```

Reply:
I thought of that, but was worried that during thread startup maybe there might be a spurious short lived period where the thread was BLOCKED. Probably not possible, but I didn't want to have to go about proving as much.